nomos-services/cryptarchia-consensus/src/states.rs

```diff
@@ -0,0 +1,132 @@
+// std
+use std::marker::PhantomData;
+// Crates
+use cl::NoteWitness;
+use overwatch_rs::services::state::ServiceState;
+use serde::{Deserialize, Serialize};
+// Internal
+use crate::leadership::Leader;
+use crate::{Cryptarchia, CryptarchiaSettings, Error};
+use nomos_core::header::HeaderId;
+use nomos_ledger::LedgerState;
+
+pub struct GenesisRecoveryStrategy {
+    pub tip: HeaderId,
+}
+
+pub struct SecurityRecoveryStrategy {
+    pub tip: HeaderId,
+    pub security_block_id: HeaderId,
+    pub security_ledger_state: LedgerState,
+    pub security_leader_notes: Vec<NoteWitness>,
+}
+
+pub enum CryptarchiaInitialisationStrategy {
+    Genesis,
+    RecoveryFromGenesis(GenesisRecoveryStrategy),
+    RecoveryFromSecurity(Box<SecurityRecoveryStrategy>),
+}
+
+#[derive(Clone, Serialize, Deserialize)]
+pub struct CryptarchiaConsensusState<TxS, BxS, NetworkAdapterSettings, BlendAdapterSettings> {
+    tip: Option<HeaderId>,
+    security_block: Option<HeaderId>,
+    security_ledger_state: Option<LedgerState>,
+    security_leader_notes: Option<Vec<NoteWitness>>,
+    _txs: PhantomData<TxS>,
+    _bxs: PhantomData<BxS>,
+    _network_adapter_settings: PhantomData<NetworkAdapterSettings>,
+    _blend_adapter_settings: PhantomData<BlendAdapterSettings>,
+}
+
+impl<TxS, BxS, NetworkAdapterSettings, BlendAdapterSettings>
+    CryptarchiaConsensusState<TxS, BxS, NetworkAdapterSettings, BlendAdapterSettings>
+{
+    pub fn new(
+        tip: Option<HeaderId>,
+        security_block: Option<HeaderId>,
+        security_ledger_state: Option<LedgerState>,
+        security_leader_notes: Option<Vec<NoteWitness>>,
+    ) -> Self {
+        Self {
+            tip,
+            security_block,
+            security_ledger_state,
+            security_leader_notes,
+            _txs: Default::default(),
```

Comment:
My Clippy recommends again:
```suggestion
            _txs: PhantomData::default(),
```